bogota-july-2018/Dominik Ortiz Galop/js/app.js

```diff
@@ -0,0 +1,139 @@
+/* ======= Model ======= */
+
+let model = {
+  number1: 0,
+  number2: 0
+};
+
+/* ======= Controller ======= */
+
+const controller = {
+  setNumber(modelNumber, value) {
+    model[modelNumber] = value;
+  },
+
+  reset() {
+    model = {
+      number1: 0,
+      number2: 0
+    };
+  },
+
+  getNumber(number) {
+    return model[number];
+  },
+
+  sum(numb1, numb2) {
+    return numb1 + numb2;
+  },
+
+  rest(numb1, numb2) {
+    return numb1 - numb2;
+  },
+
+  multiplication(numb1, numb2) {
+    return numb1 * numb2;
+  },
+
+  division(numb1, numb2) {
+    return parseFloat(numb1 / numb2);
+  },
+
+  percentage(numb1, per) {
+    return parseFloat((numb1 / 100) * per);
+  },
+
+  changeSymbol(modelNumber, symbol) {
+    const change = symbol
+      ? Math.abs(model[modelNumber])
+      : -Math.abs(model[modelNumber]);
+    model[modelNumber] = change;
+    return change;
+  }
+};
+
+/* ======= View ======= */
+
+const view = {
+  init() {
+    this.container = document.getElementById("container");
+    this.display = document.getElementById("display");
+    this.result = document.getElementById("result");
+    this.changeValue = document.getElementById("changeValue");
+    this.clear = document.getElementById("clear");
+    this.renderValue();
+  },
+
+  currentOperation(numb1, numb2, typeOfOperation) {
+    switch (typeOfOperation) {
+      case "+":
+        return controller.sum(numb1, numb2);
+      case "-":
+        return controller.rest(numb1, numb2);
+      case "/":
+        return controller.division(numb1, numb2);
+      case "x":
+        return controller.multiplication(numb1, numb2);
+      case "%":
+        return controller.percentage(numb1, numb2);
+      default:
+        return 0;
+    }
+  },
+
+  renderValue() {
+    let state = true;
+    let typeOfOperation;
+    let changeSymbol = true;
+    this.display.textContent = controller.getNumber("number1");
+    let newNumber = "";
+    this.container.addEventListener("click", e => {
+      changeSymbol = !changeSymbol;
+      switch (e.target.textContent) {
+        case "+":
+        case "-":
+        case "x":
+        case "/":
+        case "%":
+          state = !state;
```

Comment:
At this point this looks like a boolean, why is called `state`? what it actually means?